### ISSUE TITLE
Array assignment fixer generates syntax errors 

### ISSUE THREAD

**Issue by lsl** [WordPress/WordPress-Coding-Standards#1838]
## Bug Description
Array fixer generates syntax errors

## Minimal Code Snippet
```php
[$a, $b] = ['a', 'b'];
```
Fixes to:
```php
array( $a, $b ) = array( 'a', 'b' );
```
Expected
```php
list( $a, $b ) = array( 'a', 'b' );
```

## Error Code (php -l)
```
PHP Parse error:  syntax error, unexpected '=' in ...
```
## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.3.12
| PHP_CodeSniffer version | 3.5.0
| WPCS version            | develop (d45f5e5c)
| WPCS install type       | git 

## Tested Against `develop` branch?
- [x] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
This is an issue with the PHPCS `Generic.Syntax.DisallowShortArraySyntax` sniff. Please report this [upstream](https://github.com/squizlabs/php_codesniffer/).

Note: the "expected" code sample is unlikely to ever be the outcome of the fixer as the sniff is about arrays and should not touch lists at all, so I suggest using the below as "Expected":

```php
[ $a, $b ] = array( 'a', 'b' );
```

**Comment by jrfnl** [comment]
This was going to be fixed by the Utils refactor in PHPCS, but as that didn't get merged, there is no fix available upstream.

In PHPCSExtra a semi-duplicate of the PHPCS sniff will be available which does account for short lists and handles them correctly.

**Comment by sirbrillig** [comment]
> Note: the "expected" code sample is unlikely to ever be the outcome of the fixer as the sniff is about arrays and should not touch lists at all

100% agreed, although I _think_ that means that this standard needs a new sniff to disallow list shorthand because AFAIK shorthand list assignment wasn't available until PHP 7.1 and WPCS is still PHP 5.6.

**Comment by jrfnl** [comment]
> a new sniff to disallow list shorthand because AFAIK shorthand list assignment wasn't available until PHP 7.1 and WPCS is still PHP 5.6.

@sirbrillig As that's not a "hard" rule (not in the handbook), that's more something for the PHPCompatibility standard to detect, which is a [recommendation](https://github.com/WordPress/WordPress-Coding-Standards#recommended-additional-rulesets) to be used alongside of this standard.

All the same, if it would be desired to add such a sniff, PHPCSExtra contains one - `Universal.Lists.DisallowShortListSyntax` - , so it would be an easy add ;-)

As for the original issue reported here, that will be fixed in WPCS 3.0.0 and the commit to fix this is lined up to be pulled in the next round of WPCS 3.0.0 PRs.
